Makefile

```
@@ -366,29 +366,36 @@ release-compress:
 	fi
 	cd $(DIST)/release/; for file in `find . -type f -name "*"`; do echo "compressing $${file}" && gxz -k -9 $${file}; done;
 
-.PHONY: js
-js:
-	@if ([ ! -d "$(PWD)/node_modules" ]); then \
-		echo "node_modules directory is absent, please run 'npm install' first"; \
+npm-check:
+	@hash npm > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		echo "Please install Node.js 8.x or greater with npm"; \
 		exit 1; \
 	fi;
 	@hash npx > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		echo "Please install npm version 5.2+"; \
+		echo "Please install Node.js 8.x or greater with npm"; \
 		exit 1; \
 	fi;
+
+.PHONY: npm
+npm:
+	$(MAKE) npm-check
+	npm install --no-save
+
+.PHONY: npm-update
+npm-update:
+	$(MAKE) npm-check
+	npx updates -cu
+	rm -rf node_modules
+	npm install --package-lock
+
+.PHONY: js
+js:
```

Please use the make dependance management.
```suggestion
js: npm
	npx eslint public/js
```